rust/candid/src/parser/random.rs

```diff
@@ -216,3 +121,75 @@ impl IDLValue {
         })
     }
 }
+
+impl IDLArgs {
+    pub fn any(u: &mut Unstructured, env: &TypeEnv, types: &[Type]) -> Result<Self> {
+        let mut args = Vec::new();
+        for t in types.iter() {
+            let mut config = GenConfig::default();
+            let v = config.any(u, env, t)?;
+            args.push(v);
+        }
+        Ok(IDLArgs { args })
+    }
+}
+
+impl TypeEnv {
+    /// Upper bound of type size. Returns None if infinite.
+    fn size_helper(&self, seen: &mut HashSet<String>, t: &Type) -> Option<usize> {
+        use Type::*;
+        Some(match t {
+            Var(id) => {
+                if seen.insert(id.to_string()) {
+                    let ty = self.rec_find_type(id).unwrap();
+                    self.size_helper(seen, &ty)?
+                } else {
+                    return None;
+                }
+            }
```

Comment:
This returns ∞ whenever a type was seen before, but I _think_ you want to return ∞ whenever a type is recursive?
I.e.  `type a = opt bool; record { a; a }` is not infinite.

It seems you are using one mutable `seen` for the whole type, rather than an immutable that you modify only locallly as you recurse.

Reply:
Thanks for catching this! Fixed.